engine/api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessState.scala

```diff
@@ -1,21 +1,64 @@
 package pl.touk.nussknacker.engine.api.deployment
 
-import io.circe.{Decoder, Encoder}
 import io.circe.generic.JsonCodec
+import io.circe.{Decoder, Encoder, Json}
 import pl.touk.nussknacker.engine.api.ProcessVersion
+import pl.touk.nussknacker.engine.api.deployment.StateAction.StateAction
+import pl.touk.nussknacker.engine.api.deployment.StatusState.StateStatus
+import pl.touk.nussknacker.engine.defaults.deployment.DefaultProcessStateConfigurator
 
-@JsonCodec case class ProcessState(id: DeploymentId,
-                                   runningState: RunningState.Value,
+trait ProcessStateConfigurator {
+  def processStateStatus: Enumeration
+  def statusTooltips: Map[StateStatus, String]
+  def statusIcons: Map[StateStatus, String]
+  def isRunning(statusState: String): Boolean
```

Comment:
Please move it to traits that will be mixed to StateStatus. Lets this class be as simple as it can. Also IMO we can leave TODO near this traits that we should find usage of them and think if it is really needed to use it in custom implementations. Maybe we can replace them with some more explicit methods like `prepareAnnouncementMessage` and in simple implementation it will check which processes are running or any other state... but in custom one, it can return whatever - also empty string.

Reply:
I've moved it to StateStatus:
'''
object StateStatus {
  class StateStatus(val name: String) {
    def isOkForDeployed: Boolean = isDuringDeploy || isRunning
    def isDuringDeploy: Boolean = false
    def isFinished: Boolean = false
    def isRunning: Boolean = false
  }

  final class DuringDeployStateStatus(name: String) extends StateStatus(name) {
    override def isDuringDeploy: Boolean = true
  }

  final class FinishedStateStatus(name: String) extends StateStatus(name) {
    override def isFinished: Boolean = true
  }

  final class RunningStateStatus(name: String) extends StateStatus(name) {
    override def isRunning: Boolean = true
  }
}
'''